### ISSUE TITLE
Adding the cart functionality and making dedicated My Account page

### ISSUE THREAD

**Issue by anwesha2002** [MAVRICK-1/e-commerce_website#60]
**Is your feature request related to a problem? Please describe.**
I would like to work on cart functionality as when user will add something in the cart it will get updated.

**Describe the solution you'd like**
For making the cart functionality  I have to make a "add to cart" button for all the products  and whenever user click the button the cart items value will be change and on clicking on the cart icon they will be able to see the selected items through a drawer or on a page. It will also show the total price for the selected items in the end. It will also have a "+" and "-" button to change the number of products they select.

**Describe alternatives you've considered**
@MAVRICK-1 
I am a GSSOC'24 contributor. please assign this to me so that I can work on this. I have worked on similar thing previously.

**Additional context**
I would also like to work on My account section by adding a dedicated page to see users account detail and users would be able to edit.

priority :  high

**Comment by MAVRICK-1** [comment]
@anwesha2002 that's a great idea , you can divide this work into 2 parts level3 each, you can start your work after #36  this merge

**Comment by MAVRICK-1** [comment]
@anwesha2002 you can  start  your work now , add to the cars functionality is already fixed . you go on with the my account page.

**Comment by MAVRICK-1** [comment]
@anwesha2002 Do you need Firebase access to create a new database for the user? Currently, we are retrieving user data from Firebase Authentication. If possible, could you create a form for user details and store that information in Firestore? We can divide the task into three parts, each at level 3

**Comment by MAVRICK-1** [comment]
### Task: Implement User Details Form and Store Data in Firestore

#### Part 1: User Interface (UI)

1. **Create User Details Form:**
   - Design a form to collect user details such as name, email, address, etc.
   - Ensure the form is user-friendly and intuitive.
   - Include appropriate input fields, labels, and validation where necessary.

2. **UI for Editing User Details:**
   - Provide a way for users to edit their details.
   - Design an interface that allows users to easily update their information.
   - Consider using modal dialogs or dedicated edit pages for editing user details.

#### Part 2: Storing Data

1. **Firebase Integration:**
   - Set up Firebase Firestore in your Angular project if not already done.
   - Configure Firebase authentication to authenticate users.

2. **Store User Details:**
   - Implement functionality to store user details collected from the form into Firestore.
   - Ensure proper validation of data before storing it in the database.
   - Use Firestore API to write data to the database.

#### Part 3: Editing Functionality

1. **Retrieve User Details:**
   - Implement functionality to fetch user details from Firestore.
   - Display the fetched user details in the form for editing.

2. **Update User Details:**
   - Allow users to make changes to their details.
   - Implement functionality to update the user details in Firestore when the user submits the updated form.
   - Ensure data integrity and security by validating and sanitizing user input before updating the database.

3. **Error Handling:**
   - Implement error handling mechanisms to handle cases such as network errors or database write failures.
   - Provide appropriate feedback to users in case of errors during data submission or retrieval.

**Comment by anwesha2002** [comment]
@MAVRICK-1 okay I will do this one by one and will make a PR before preceding to every next part

**Comment by MAVRICK-1** [comment]
@anwesha2002 will store to user data in firebase ?

**Comment by anwesha2002** [comment]
@MAVRICK-1 yeah no issue I will store the data in firebase

**Comment by anwesha2002** [comment]
@MAVRICK-1 for that I will need firebase access I started working on the users form once it done I will need access for user database

**Comment by anwesha2002** [comment]
@MAVRICK-1   anweshadas909@gmail.com

**Comment by MAVRICK-1** [comment]
@anwesha2002 check your email

**Comment by anwesha2002** [comment]
@MAVRICK-1 I solved all the 3 parts it is storing data in firebase and the form is editable and editable .

**Comment by anwesha2002** [comment]
@MAVRICK-1 please LMK for any issue
